### PR TITLE
Decompose models

### PR DESCRIPTION
**Pull request by FalseR20**

### CODE REVIEW

**Review by AlexandrZagorskiy — APPROVED** [review]

File: tests/api_test.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def for_each_tripple_func(src: ScAddr, edge: ScAddr, trg: ScAddr):
         )
         self.get_server_message('{"errors": [], "id": 1, "event": false, "status": true, "payload": ' + payload + "}")
         templ = ScTemplate()
-        templ.triple([ScAddr(0), "_class_node"], sc_types.EDGE_ACCESS_VAR_POS_PERM, ScAddr(0))
+        templ.triple((ScAddr(0), "_class_node"), sc_types.EDGE_ACCESS_VAR_POS_PERM, ScAddr(0))
```
Comment (NikitaZotov) [inline tests/api_test.py]: Can you use square brackets now?

File: tests/api_test.py [hunk 2]
```diff
@@ -410,14 +410,14 @@ def test_wrong_template(self):
...
-            templ.triple([ScAddr(0), "_class_node"], ScAddr(0), [sc_types.LINK_CONST, "_const_link"])
+            templ.triple((ScAddr(0), "_class_node"), ScAddr(0), (sc_types.LINK_CONST, "_const_link"))
 
         with pytest.raises(InvalidTypeError):
             templ = ScTemplate()
             templ.triple_with_relation(
-                [ScAddr(0), "_main_node"],
-                [sc_types.EDGE_D_COMMON_CONST, "_const_edge"],
-                [sc_types.LINK_VAR, "_link"],
+                ScAddr(0) >> "_main_node",
```
Comment (NikitaZotov) [inline tests/api_test.py]: Don't change tests. Create other ones

Comment (FalseR20) [reply tests/api_test.py]: __rshift__ will be added in other PR #45

File: tests/api_test.py (hunk 1 above)
Comment (FalseR20) [reply tests/api_test.py]: Yes, but with deprecated warning

File: src/sc_client/models/sc_construction.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+class ScIdtfResolveParams(TypedDict):
+    idtf: str
+    type: ScType | None
+
+
+class ScLinkContentType(Enum):
+    INT = 0
+    # FLOAT = 1
```
Comment (NikitaZotov) [inline src/sc_client/models/sc_construction.py]: Why was it comment?

File: src/sc_client/models/sc_template.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+    def triple_with_relation(
+        self,
+        src: ScTemplateParam,
+        edge: ScTemplateParam,
+        trg: ScTemplateParam,
+        edge2: ScTemplateParam,
+        src2: ScTemplateParam,
+    ) -> None:
+        if not isinstance(edge, (tuple, list)):  # TODO: remove list support in version 0.3.0
```
Comment (NikitaZotov) [inline src/sc_client/models/sc_template.py:73]: It is better to deprecate, don't remove

File: tests/api_test.py (hunk 1 above)
Comment (NikitaZotov) [reply tests/api_test.py]: Then leave tests with this brackets or add tests with this brackets to show that we didn't remove this logic case, but not remove

Answer by @FalseR20: there are 2 types of brackets in tests

File: README.md [hunk 5]
```diff
@@ -238,7 +238,7 @@ template = keynodes['my_template']
 search_results = client.template_search(template)
 search_result = search_results[0]
 
-search_result.size()
+len(search_result)
```
Comment (NikitaZotov) [inline README.md:245]: Don't remove this logic. Leave it and add your own one if it is matter

File: README.md [hunk 6]
```diff
@@ -223,13 +223,13 @@ search_params = {'_link': link_node, '_var_node': 'node_idtf'} # value element c
...
 search_result = search_results[0]
 
-search_result.size() # count of elements in the resulting construction
-search_result.get(0).value # get an element from the result by index
+len(search_result) # count of elements in the resulting construction
+search_result[0].value # get an element from the result by index
 search_result.get('_class_node').value # get an element from the result by alias
 
-def for_each_tripple_func(src: ScAddr, edge: ScAddr, trg: ScAddr):
+for src, edge, trg in search_result:
```
Comment (NikitaZotov) [inline README.md:232]: Leave for_each_tripple_func, don't remove it. Add tests for it

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: src/sc_client/models/sc_construction.py (hunk 3 above)
Comment (FalseR20) [reply src/sc_client/models/sc_construction.py]: It doesn't work

File: src/sc_client/models/sc_template.py (hunk 4 above)
Comment (FalseR20) [reply src/sc_client/models/sc_template.py:73]: Next release is 0.2.5, in 0.3.0 it will be removed

File: README.md (hunk 5 above)
Comment (FalseR20) [reply README.md:245]: In versions 0.2.X I won't remove it

File: README.md (hunk 6 above)
Comment (FalseR20) [reply README.md:232]: I changed `for_each_tripple_func` tests to for-statement tests

File: tests/api_test.py (hunk 1 above)
Comment (AlexandrZagorskiy) [reply tests/api_test.py]: How long we should support deprecated methods? As I understand, we should remove every deprecated method in a next complete version, like 0.3.0? 
So, in the 0.2.5 release we will have bunch of deprecated methods with new ones. Should we discuss it in a separate discussion?

Comment (FalseR20) [reply tests/api_test.py]: I decided so myself. I thinks it would be great

Comment (FalseR20) [reply tests/api_test.py]: I added deprecated tests

File: src/sc_client/models/sc_construction.py (hunk 3 above)
Comment (NikitaZotov) [reply src/sc_client/models/sc_construction.py]: Can you describe what doesn't work?

File: src/sc_client/models/sc_template.py (hunk 4 above)
Comment (NikitaZotov) [reply src/sc_client/models/sc_template.py:73]: okay

File: README.md (hunk 5 above)
Comment (NikitaZotov) [reply README.md:245]: But documentation have been changed. And tests so have been changed

File: README.md (hunk 6 above)
Comment (NikitaZotov) [reply README.md:232]: I propose do not remove it in this release

Comment (FalseR20) [reply README.md:232]: Yeah, sure. I didn't remove

File: src/sc_client/models/sc_construction.py (hunk 3 above)
Comment (FalseR20) [reply src/sc_client/models/sc_construction.py]: There were problems with float and binary. Maybe smth changed

File: README.md (hunk 5 above)
Comment (FalseR20) [reply README.md:245]: I'll add deprecated in docs

File: src/sc_client/models/sc_construction.py (hunk 3 above)
Comment (FalseR20) [reply src/sc_client/models/sc_construction.py]: Okay, I checked, and float also works. But bytes don't. Updated

File: README.md (hunk 5 above)
Comment (FalseR20) [reply README.md:245]: Added. Also I think we need refactor docs, there is no everything

**Review by AlexandrZagorskiy — APPROVED** [review]

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**